hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/BaseRollbackActionExecutor.java

```diff
@@ -155,12 +156,13 @@ private void validateRollbackCommitSequence() {
     // since with LAZY rollback we support parallel writing which can allow a new inflight while rollback is ongoing
     // Remove this once we support LAZY rollback of failed writes by default as parallel writing becomes the default
     // writer mode.
-    if (config.getFailedWritesCleanPolicy().isEager()) {
+    if (config.getFailedWritesCleanPolicy().isEager()  && !HoodieTableMetadata.isMetadataTable(config.getBasePath())) {
```

Comment:
with regular data table, w/ eager rollbacks, we have a constraint that commit to be rolledback has to be the last commit in timeline. But w/ metadata table, since its a replication of data table commits, we can't guarantee the order of commits and so we can't honor the same constraint and had to relax it. Added a java doc on L 165.

Reply:
So we just loosen restrictions for MDT rollback check.